modules/apps/object/object-web/src/main/resources/META-INF/resources/js/components/ObjectAction/Actions.tsx

```diff
@@ -92,6 +93,67 @@ export default function Actions({
 		);
 	}
 
+	function objectActionSourceDataRenderer({itemData}: {itemData: ItemData}) {
```

Comment:
@seltong @G-Sa We can declare this function outside/before the Actions component to avoid getting it recreated every time the component is rendered, since it only depends on the info passed as parameter. Could you please also move all the other DataRenderers, please? Thanks!

Reply:
Another thing, please capitalize the name of this function. We have this pattern for all functions that return JSX/React components. Please, check if this applies for the other DataRenderers as well.

Reply:
@naraandrade The `ObjectActionLabelDataRenderer` is using the `url` parameter of the Actions component. I couldn't find a way to isolate this function and move this to outside/before the Actions component, do you have any suggestion?

Reply:
@seltong You could receive the url through the itemData, but it may take some work and introduce small issues. Since this PR is High Priority, just leave `ObjectActionLabelDataRenderer` inside the Actions component for now.